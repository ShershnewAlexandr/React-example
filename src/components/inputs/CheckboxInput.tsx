import React from 'react';
import { Grid } from '@material-ui/core';
import newId from '../../utils/newid';
import './InputsStyles.css';

interface CheckboxInputProps {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  label: string;
  disabled: boolean;
}

function CheckboxInput({ label, onChange, disabled }: CheckboxInputProps) {
  const id = newId();

  return (
    <Grid container direction="column" justify="center" alignItems="flex-start">
      <Grid item>
        <label htmlFor={id}>{label}</label>
      </Grid>
      <Grid item>
        <input
          id={id}
          type="checkbox"
          onChange={onChange}
          disabled={disabled}
        />
      </Grid>
    </Grid>
  );
}

export default CheckboxInput;
