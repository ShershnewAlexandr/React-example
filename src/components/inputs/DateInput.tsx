import React from 'react';
import { WrappedFieldProps } from 'redux-form';
import { useTranslation } from 'react-i18next';
import { Grid } from '@material-ui/core';
import newId from '../../utils/newid';
import './InputsStyles.css';

interface DateInputProps {
  label: string;
  disabled: boolean;
}

function DateInput({
  label,
  disabled,
  input,
  meta,
}: DateInputProps & WrappedFieldProps) {
  const id = newId();
  const { invalid, touched, error } = meta;
  const { t } = useTranslation();

  return (
    <Grid container>
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        type="date"
        {...input}
        className={invalid && touched ? 'text-input-error' : ''}
        disabled={disabled}
      />
      {invalid && touched && <div className="error">{t(error)}</div>}
    </Grid>
  );
}

export default DateInput;
