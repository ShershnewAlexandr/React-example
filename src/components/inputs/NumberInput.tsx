import React from 'react';
import { WrappedFieldProps } from 'redux-form';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import newId from '../../utils/newid';
import './InputsStyles.css';

interface NumberInputProps {
  label: string;
  min: number;
  max: number;
  step: number;
  disabled: boolean;
}

function NumberInput(props: NumberInputProps & WrappedFieldProps) {
  const id = newId();
  const { label, min, max, step, input, meta, disabled } = props;
  const { invalid, touched, error } = meta;
  const { t } = useTranslation();

  return (
    <Grid container>
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        type="number"
        {...input}
        min={min}
        max={max}
        step={step}
        className={invalid && touched ? 'text-input-error' : ''}
        disabled={disabled}
      />
      {invalid && touched && <div className="error">{t(error)}</div>}
    </Grid>
  );
}

export default NumberInput;
