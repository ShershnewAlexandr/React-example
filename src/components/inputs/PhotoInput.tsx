import React from 'react';
import ReactFileReader from 'react-file-reader';
import { Grid } from '@material-ui/core';
import './InputsStyles.css';
import './PhotoInputStyles.css';

interface PhotoInputProps {
  label: string;
  prerenderPhotoURL: string;
  disabled: boolean;
  onClickAction: (files: ReactFileReader, url: string) => void;
}

function PhotoInput(props: PhotoInputProps) {
  function onChangeFile(files: ReactFileReader) {
    const url = URL.createObjectURL(files.fileList[0]);
    props.onClickAction(files, url);
  }

  const { label, prerenderPhotoURL, disabled } = props;

  return (
    <Grid container direction="column">
      <Grid item>
        <label>{label}</label>
      </Grid>
      <Grid item>
        <div
          className="photoButton"
          style={{ backgroundImage: `url(${prerenderPhotoURL})` }}
        >
          {!disabled && (
            <ReactFileReader handleFiles={onChangeFile} base64>
              <div
                className={
                  prerenderPhotoURL
                    ? 'photoButtonPlus photoButtonPlusWithPhoto'
                    : 'photoButtonPlus'
                }
              >
                +
              </div>
            </ReactFileReader>
          )}
        </div>
      </Grid>
    </Grid>
  );
}

export default PhotoInput;
