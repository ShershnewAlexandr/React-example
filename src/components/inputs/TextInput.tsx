import React from 'react';
import { WrappedFieldProps } from 'redux-form';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import newId from '../../utils/newid';
import './InputsStyles.css';

interface TextInputProps {
  label: string;
  placeholder: string;
  disabled: boolean;
  isPassword: boolean;
}

function TextInput(props: TextInputProps & WrappedFieldProps) {
  const id = newId();
  const { label, placeholder, disabled, isPassword, input, meta } = props;
  const { invalid, touched, error } = meta;
  const { t } = useTranslation();

  return (
    <Grid container item>
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        type={isPassword ? 'password' : 'text'}
        {...input}
        className={invalid && touched ? 'text-input-error' : ''}
        placeholder={placeholder}
        disabled={disabled}
      />
      {invalid && touched && <div className="error">{t(error)}</div>}
    </Grid>
  );
}

export default TextInput;
