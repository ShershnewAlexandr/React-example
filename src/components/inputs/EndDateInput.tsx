import React from 'react';
import { WrappedFieldProps } from 'redux-form';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import DateInput from './DateInput';
import CheckboxInput from './CheckboxInput';
import './InputsStyles.css';

interface EndDateInputProps {
  disabled: boolean;
  checked: boolean;
  checkboxChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  label: string;
}

function EndDateInput(props: EndDateInputProps & WrappedFieldProps) {
  const { t } = useTranslation();
  const { disabled, checked, checkboxChange } = props;
  return (
    <Grid
      container
      direction="row"
      justify="flex-end"
      alignItems="center"
      spacing={2}
    >
      <Grid item xs={10}>
        <DateInput {...props} disabled={checked || disabled} />
      </Grid>
      <Grid item xs={2}>
        <CheckboxInput
          onChange={checkboxChange}
          label={t('education.now')}
          disabled={disabled}
        />
      </Grid>
    </Grid>
  );
}

export default EndDateInput;
