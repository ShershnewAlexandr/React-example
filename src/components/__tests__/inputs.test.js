import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import CheckboxInput from '../inputs/CheckboxInput';
import DateInput from '../inputs/DateInput';
import EndDateInput from '../inputs/EndDateInput';
import NumberInput from '../inputs/NumberInput';
import PhotoInput from '../inputs/PhotoInput';
import TextInput from '../inputs/TextInput';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('inputs', () => {
  const CheckboxInputProps = {
    onChange: jest.fn(),
    label: 'label',
    disabled: false,
  };

  const DateInputProps = {
    meta: { invalid: false, touched: false, error: null },
    label: 'label',
    disabled: false,
  };

  const EndDateInputProps = {
    disabled: false,
    checked: false,
    checkboxChange: jest.fn(),
    label: 'label',
  };

  const NumberInputProps = {
    meta: { invalid: false, touched: false, error: null },
    label: 'label',
    min: 0,
    max: 5,
    step: 1,
    disabled: false,
  };

  const PhotoInputProps = {
    label: 'label',
    prerenderPhotoURL: 'url',
    disabled: 'false',
    onClickAction: jest.fn(),
  };

  const TextInputProps = {
    meta: { invalid: false, touched: false, error: null },
    label: 'label',
    placeholder: 'placeholder',
    disabled: false,
    isPassword: false,
  };

  const checkboxInput = shallow(<CheckboxInput {...CheckboxInputProps} />);
  const dateInput = shallow(<DateInput {...DateInputProps} />);
  const endDateInput = shallow(<EndDateInput {...EndDateInputProps} />);
  const numberInput = shallow(<NumberInput {...NumberInputProps} />);
  const photoInput = shallow(<PhotoInput {...PhotoInputProps} />);
  const textInput = shallow(<TextInput {...TextInputProps} />);

  it('CheckboxInput snapshot', () => {
    expect(toJson(checkboxInput)).toMatchSnapshot();
  });

  it('DateInput snapshot', () => {
    expect(toJson(dateInput)).toMatchSnapshot();
  });

  it('EndDateInput snapshot', () => {
    expect(toJson(endDateInput)).toMatchSnapshot();
  });

  it('NumberInput snapshot', () => {
    expect(toJson(numberInput)).toMatchSnapshot();
  });

  it('PhotoInput snapshot', () => {
    expect(toJson(photoInput)).toMatchSnapshot();
  });

  it('TextInput snapshot', () => {
    expect(toJson(textInput)).toMatchSnapshot();
  });

  it('CheckboxInput should call onchange', () => {
    checkboxInput
      .find('input')
      .simulate('change')
      .simulate('change');
    expect(CheckboxInputProps.onChange.mock.calls.length).toBe(2);
  });
});
