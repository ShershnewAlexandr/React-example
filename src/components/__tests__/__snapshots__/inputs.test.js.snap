// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputs CheckboxInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="flex-start"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <label
      htmlFor="specialId1235"
    >
      label
    </label>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <input
      disabled={false}
      id="specialId1235"
      onChange={[MockFunction]}
      type="checkbox"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`inputs DateInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <label
    htmlFor="specialId1236"
  >
    label
  </label>
  <input
    className=""
    disabled={false}
    id="specialId1236"
    type="date"
  />
</WithStyles(ForwardRef(Grid))>
`;

exports[`inputs EndDateInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="row"
  justify="flex-end"
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <DateInput
      checkboxChange={[MockFunction]}
      checked={false}
      disabled={false}
      label="label"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <CheckboxInput
      disabled={false}
      label="education.now"
      onChange={[MockFunction]}
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`inputs NumberInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <label
    htmlFor="specialId1237"
  >
    label
  </label>
  <input
    className=""
    disabled={false}
    id="specialId1237"
    max={5}
    min={0}
    step={1}
    type="number"
  />
</WithStyles(ForwardRef(Grid))>
`;

exports[`inputs PhotoInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <label>
      label
    </label>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <div
      className="photoButton"
      style={
        Object {
          "backgroundImage": "url(url)",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`inputs TextInput snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  item={true}
>
  <label
    htmlFor="specialId1238"
  >
    label
  </label>
  <input
    className=""
    disabled={false}
    id="specialId1238"
    placeholder="placeholder"
    type="text"
  />
</WithStyles(ForwardRef(Grid))>
`;
